src/interpreter/ByteCodeGenerator.cpp

```diff
@@ -478,6 +478,11 @@ ByteCodeBlock* ByteCodeGenerator::generateByteCode(Context* c, InterpretedCodeBl
                 assignStackIndexIfNeeded(cd->m_objectRegisterIndex, stackBase, stackBaseWillBe, stackVariableSize);
                 break;
             }
+            case GetIteratorOpcode: {
```

Comment:
Could you explain me why only one of the new opcodes need `assignStackIndexIfNeeded` operation? Probably the code is good in the current status as well, but it's a good possibility to ask this question. :)

Reply:
There are 3 kinds of values allocated in the stack. 
- temporal value (used for bytecode interpretation)
- identifier
- literal value

The shape of stack would be sequentially allocated like below.
| temporal value | identifier | literal value |

When we generate bytecodes, we allocate and categorize each value to the register(stack) as follow.
- temporal value : index
- identifier : `REGULAR_REGISTER_LIMIT` + index
- literal value : `REGULAR_REGISTER_LIMIT` + `VARIABLE_LIMIT` + index

After generating the bytecodes, then each allocated register should be rearranged to be allocated sequentially in the stack. So, `assignStackIndexIfNeeded` is called for each allocated register finally.

Reply:
Thank you, it will probably be useful later :)